backend/src/test/java/com/github/alaajouri/backend/service/MongoUserDetailsServiceTest.java

```diff
@@ -0,0 +1,48 @@
+package com.github.alaajouri.backend.service;
+
+import com.github.alaajouri.backend.model.MongoUser;
+import com.github.alaajouri.backend.repository.MongoUserRepository;
+import org.junit.jupiter.api.Test;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.*;
+
+
+class MongoUserDetailsServiceTest {
+    PasswordEncoder passwordEncoder;
+    MongoUserDetailsService mongoUserDetailsService;
+    MongoUserRepository mongoUserRepository;
+    IdGenerator idGenerator;
+    MongoUser mongoUser;
+
+    @BeforeEach
+    void setUp() {
+        mongoUserRepository = mock(MongoUserRepository.class);
+        passwordEncoder = mock(PasswordEncoder.class);
+        idGenerator = mock(IdGenerator.class);
+        mongoUserDetailsService = new MongoUserDetailsService(mongoUserRepository);
+        mongoUser = new MongoUser("1", "username", "123", "BASIC");
+    }
+
+    @Test
+    void loadUserByUsername() {
```

Comment:
Here you are not covering the case if a user is not found and an exception is thrown. You could add an additional test for that.